eravm-stable-interface/src/tracer_interface.rs

```diff
@@ -53,18 +53,19 @@ macro_rules! pub_struct {
 }
 
 pub mod opcodes {
+    use super::{CallingMode, ReturnType};
+    use std::marker::PhantomData;
+
     forall_simple_opcodes!(pub_struct);
-    pub struct FarCall<M: TypeLevelCallingMode>(M);
-    pub struct Ret<T: TypeLevelReturnType>(T);
```

Comment:
Does using PhantomData silence the warning about these not being used? If not, it is unnecessary.

Reply:
Yep, it does; this is exactly why I've switched to it. Weirdly enough, these warnings aren't emitted in this repo, only when `vm2` is compiled in the Era codebase. Haven't figured out why yet.